rpcs3/Emu/RSX/Overlays/overlay_message_dialog.cpp

```diff
@@ -240,13 +240,20 @@ namespace rsx
 					}
 					else
 					{
+						if (m_waiters.add_fetch(1) <= 1)
```

Comment:
It would be nice if these things were explained in the code

Reply:
Simplified the whole thing to just track the threads themselves.
This change to std::thread means the worker threads can't have names for now though since dealing with named_thread<F> requires creating a new class for that and I just don't want to do too much for a fixup.